### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by DidenkoDenis306**
Some tests have failed, but it works correctly (checked in Postman)

### CODE REVIEW

File: src/routes/expenses.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+
+  if (userId) {
+    expenses = expenses.filter(expense => (
+      expense.userId === Number(userId)));
```
Comment (33Orange) [inline src/routes/expenses.js:19]:
```suggestion
    expenses = expenses.filter(expense => (
      expense.userId === Number(userId)
    ));
```

File: src/routes/expenses.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    ));
+  }
+
+  if (from && to) {
+    expenses = expenses.filter(expense => (
+      expense.spentAt >= from && expense.spentAt <= to
+    ));
+  }
+
+  res.send(expenses);
```
Comment (33Orange) [inline src/routes/expenses.js:46]: Same about json in a response

File: src/routes/expenses.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  const findUser = users.find(user => user.id === Number(userId));
+
+  if (!findUser) {
+    res.sendStatus(400);
+
```
Comment (33Orange) [inline src/routes/expenses.js:63]: Same about the message in the errors

File: src/routes/expenses.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  res.statusCode = 201;
+  res.send(newExpense);
+});
+
+router.get('/:expenseId', (req, res) => {
+  const { expenseId } = req.params;
+
+  const findExpense = expenses.find(expense => (
+    expense.id === Number(expenseId)));
```
Comment (33Orange) [inline src/routes/expenses.js:87]:
```suggestion
  const findExpense = expenses.find(expense => (
    expense.id === Number(expenseId)
  ));
```

File: src/routes/expenses.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  }
+
+  res.send(findExpense);
+});
+
+router.patch('/:expenseId', (req, res) => {
+  const { expenseId } = req.params;
+
+  const findExpense = expenses.find(expense => (
+    expense.id === Number(expenseId)));
```
Comment (33Orange) [inline src/routes/expenses.js:102]:
```suggestion
  const findExpense = expenses.find(expense => (
    expense.id === Number(expenseId)
  ));
```

File: src/routes/expenses.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  Object.assign(findExpense, updatedExpense);
+
+  res.send(findExpense);
+});
+
+router.delete('/:expenseId', (req, res) => {
+  const { expenseId } = req.params;
+
+  const filteredExpenses = expenses.filter(expense => (
+    expense.id !== Number(expenseId)));
```
Comment (33Orange) [inline src/routes/expenses.js:124]:
```suggestion
  const filteredExpenses = expenses.filter(expense => (
    expense.id !== Number(expenseId)
  ));
```

**Review by 33Orange — CHANGES_REQUESTED** [review]
Almost done

A few moments.
Please wrap all your responses in a JSON and add some messages to the errors

**Review by naziamoff — CHANGES_REQUESTED** [review]
it is super difficult to read and maintain if all the logic is kept in a single file
Let's separate logic into different layers

File: src/routes/users.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    res.status(400);
+    res.send('User is not added');
+
+    return;
+  }
+
+  const newUser = {
+    id: Math.floor(Math.random() * 100),
+    name,
+  };
```
Comment (33Orange) [inline src/routes/users.js:25]: Check @naziamoff comment about the layouts.

Create 3 layouts. Routes, Controllers, Services.

For example:

**routes/users.js**

`router.post('/', userController.createUser);`

**controller/users.js**

```
const createUser = (req, res) => {
  const { name } = req.body;

  if (!name) {
    res.status(400);
    res.send('User is not added');

    return;
  }
  const newUser = userService.createUser(name);

  res.statusCode = 201;
  res.json(newUser);
}
```

And in service, you need to add the logic of interaction with the database.
Same in all other cases

**Review by 33Orange — CHANGES_REQUESTED** [review]

**Pull request closed without merging**